src/main/java/pl/xkoem/gamestates/Init.java

```diff
@@ -37,10 +38,87 @@ public GameConfiguration askForConfiguration(GameConfiguration gameConfiguration
 
     private void askForLanguage() {
         userOutput.accept("Wybierz jezyk: "); //todo obsluga wprowadzania jezykow
-
     }
 
     private void askForBoardSettings() {
-        userOutput.accept("Podaj rozmiary stolu: "); //todo obsluga rozmiarow
+        int width = askForWidth();
+        int height = askForHeight();
+        int minSymbolsToWin = askForMinSymbolsToWin(Math.min(width, height));
+        gameConfiguration.setBoardSettings(width, height, minSymbolsToWin);
     }
+
+    private int askForMinSymbolsToWin(int maxSymbolsToWin) {
+        maxSymbolsToWin = Math.max(maxSymbolsToWin, gameConfiguration.getMinSymbolsToWin());
+
+        Optional<Integer> symbolsToWin = Optional.empty();
+        while(!symbolsToWin.isPresent()) {
+            userOutput.accept("Podaj min ilosc symboli do wygrania");
+            String userData = userInput.get();
+            symbolsToWin = tryChangeStringToIntWithMinAndMaxValue(userData, gameConfiguration.getMinSymbolsToWin() , maxSymbolsToWin);
+        }
+        return symbolsToWin.get();
+    }
+
+    private int askForWidth() {
+        Optional<Integer> width = Optional.empty();
+        while(gameConfiguration.getMinX() > width.orElse(0)) {
+            userOutput.accept("Podaj szerokosc planszy");
+            String userData = userInput.get();
+            width = tryChangeStringToIntWithMinValue(userData, gameConfiguration.getMinX());
+        }
+        return width.get();
+    }
+
+    private int askForHeight() {
+        Optional<Integer> height = Optional.empty();
+        while(gameConfiguration.getMinY() > height.orElse(0)) {
+            userOutput.accept("Podaj wysokosc planszy");
+            String userData = userInput.get();
+            height = tryChangeStringToIntWithMinValue(userData, gameConfiguration.getMinY());
+        }
+        return height.get();
+    }
+
+    Optional<Integer> tryChangeStringToIntWithMinValue(String string, int minValue) {
+
+        Optional<Integer> value = tryChangeStringToInt(string);
+        if (!value.isPresent()) {
+            return value;
+        }
+        if(value.get() < minValue) {
+            userOutput.accept("Za mało ziom");
+            return Optional.empty();
+        }
+        return value;
+    }
+
+    Optional<Integer> tryChangeStringToIntWithMinAndMaxValue(String string, int minValue, int maxValue) {
+        Optional<Integer> value = tryChangeStringToInt(string);
```

Comment:
Copy paste z 82 lini?

Reply:
jak masz pomysl jak sie tego pozbyc to zapraszam xD